### PR TITLE
Update SilenceDiscarder

### PR DESCRIPTION
**Pull request by balins**
- adjusted `RTP.SilenceDiscarder` to the recent changes in RTP header extensions
- added `RTP.Header.Extension.find/2`
- updated moduledocs mentioning header extensions
- changed defaults in `RTP.SilenceDiscarder` options

Default for `silence_threshold` has been changed to quite arbitrary `64`, as this value seems to do pretty well. Previously set `127` was too strict to ever drop any packet 🤔

### CODE REVIEW

File: lib/membrane/rtp/silence_discarder.ex [hunk 1]
```diff
@@ -35,15 +30,15 @@ defmodule Membrane.RTP.SilenceDiscarder do
               ],
               silence_threshold: [
                 spec: 1..127,
-                default: 127,
+                default: 64,
```
Comment (Qizot) [inline lib/membrane/rtp/silence_discarder.ex:33]: Why did you change that to 64? It can drop the normal speech then, 127 means silent.

File: lib/membrane/rtp/silence_discarder.ex [hunk 2]
```diff
@@ -80,30 +70,26 @@ defmodule Membrane.RTP.SilenceDiscarder do
...
-          dropped > 0 ->
-            stop_dropping(buffer, state)
-
-          true ->
-            {{:ok, buffer: {:output, buffer}}, state}
-        end
-
-      _header ->
+    maybe_vad = Header.Extension.find(buffer, state.vad_id)
+    handle_vad(buffer, maybe_vad, state)
```
Comment (Qizot) [inline lib/membrane/rtp/silence_discarder.ex:74]:
```suggestion
    buffer
    |> Header.Extension.find(state.vad_id)
    |> handle_vad(buffer, state)
```

File: lib/membrane/rtp/silence_discarder.ex [hunk 3]
```diff
@@ -80,30 +70,26 @@ defmodule Membrane.RTP.SilenceDiscarder do
...
+  end
+
+  defp handle_vad(buffer, nil, state), do: {{:ok, buffer: {:output, buffer}}, state}
+
+  defp handle_vad(buffer, vad, state) do
+    %{dropped: dropped, silence_threshold: silence_threshold} = state
+    <<_v::1, audio_level::7>> = vad.data
+
+    silent? = audio_level >= silence_threshold
```
Comment (Qizot) [inline lib/membrane/rtp/silence_discarder.ex:83]: Put that directly into the `cond`

**Review by Qizot — CHANGES_REQUESTED** [review]

File: lib/membrane/rtp/silence_discarder.ex (hunk 1 above)
Comment (balins) [reply lib/membrane/rtp/silence_discarder.ex:33]: I explained in PR description, 127 would not drop anything in normal cases as there is always some ambient noise. Do we want to target only explicitly muted streams? If so, we can change it back. If we want to also consider some amount of ambient noise, we can think about a more safe default value.

Comment (Qizot) [reply lib/membrane/rtp/silence_discarder.ex]: I think that for now we are good with dropping just the ambient noise (unless you want to spend more time experimenting with what audio level can be considered silence). The main case is when there are a lot of people in the room and most of them are muted but the packets still get sent, it is pretty much the main reason for this filter to exist.

**Review by Qizot — APPROVED** [review]
